From bc31ccf29c0288906a4c352b119d1cd5cc1bcdb1 Mon Sep 17 00:00:00 2001
From: darodi <4682830+darodi@users.noreply.github.com>
Date: Tue, 10 Jan 2023 22:00:35 +0100
Subject: [PATCH] [FR - FRANCETV] Fix live for FranceInfo  - part2

---
 resources/lib/resolver_proxy.py | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/resources/lib/resolver_proxy.py b/resources/lib/resolver_proxy.py
index 5e6edd8b..0d936ec5 100644
--- a/resources/lib/resolver_proxy.py
+++ b/resources/lib/resolver_proxy.py
@@ -438,12 +438,8 @@ def get_mtvnservices_stream(plugin,
 def get_francetv_video_stream(plugin,
                               id_diffusion,
                               download_mode=False):
-    geoip_value = web_utils.geoip()
-    if not geoip_value:
-        geoip_value = 'FR'
-    resp = urlquick.get(URL_FRANCETV_PROGRAM_INFO % (id_diffusion, geoip_value),
-                        max_age=-1)
-    json_parser = resp.json()
+
+    json_parser = get_francetv_program_info(id_diffusion)
 
     if 'video' not in json_parser:
         plugin.notify('ERROR', plugin.localize(30716))
@@ -516,6 +512,15 @@ def get_francetv_video_stream(plugin,
 
 def get_francetv_live_stream(plugin, broadcast_id):
 
+    json_parser_live_id = get_francetv_program_info(broadcast_id)
+    final_url = json_parser_live_id['video']['token']
+
+    resp = urlquick.get(final_url, max_age=-1)
+    video_url = json.loads(resp.text)['url']
+    return get_stream_with_quality(plugin, video_url, license_url=URL_LICENSE_FRANCETV)
+
+
+def get_francetv_program_info(video_id):
     # Move Live TV on the new API
     geoip_value = web_utils.geoip()
     if not geoip_value:
@@ -525,13 +530,9 @@ def get_francetv_live_stream(plugin, broadcast_id):
         'browser': 'firefox',
         'device_type': 'desktop'
     }
-    resp = urlquick.get(URL_FRANCETV_PROGRAM_INFO % broadcast_id, params=params, headers=GENERIC_HEADERS, max_age=-1)
+    resp = urlquick.get(URL_FRANCETV_PROGRAM_INFO % video_id, params=params, headers=GENERIC_HEADERS, max_age=-1)
     json_parser_live_id = json.loads(resp.text)
-    final_url = json_parser_live_id['video']['token']
-
-    resp = urlquick.get(final_url, max_age=-1)
-    video_url = json.loads(resp.text)['url']
-    return get_stream_with_quality(plugin, video_url, license_url=URL_LICENSE_FRANCETV)
+    return json_parser_live_id
 
 
 # Arte Part
